shared/src/business/entities/cases/Case.sortByDocketNumber.test.ts

```diff
@@ -65,3 +65,111 @@ describe('sortByDocketNumber', () => {
     ]);
   });
 });
+
+describe('sortByDocketNumberAndGroupConsolidatedCases', () => {
+  it('should return the cases sorted properly, with a consolidated case appearing after its lead cases', () => {
+    const result = Case.sortByDocketNumberAndGroupConsolidatedCases([
+      {
+        docketNumber: '100-19',
+      },
+      {
+        docketNumber: '100-20',
+      },
+      {
+        docketNumber: '101-20', // Comes after 100-20
+        leadDocketNumber: '100-19', // But goes with 100-19
+      },
+      {
+        docketNumber: '100-18',
+      },
+    ]);
+
+    expect(result).toMatchObject([
+      { docketNumber: '100-18' },
+      {
+        docketNumber: '100-19',
+      },
+      {
+        docketNumber: '101-20',
+      },
+      {
+        docketNumber: '100-20',
+      },
+    ]);
+  });
+
+  it('should return the cases sorted properly, when cases contains a consolidated grouping and the lead case is missing', () => {
```

Comment:
The next two tests come almost directly from `shared/src/business/utilities/trialSession/getFormattedTrialSessionDetails.compareCasesByDocketNumber.test.ts`, which tests the function which my new sort function is meant to replace. (I made one small change to put the two member cases out of order in the second test.)